addon/index.js

```diff
@@ -18,15 +18,32 @@ export default class Resolver {
 
   moduleNameForFullName(fullName) {
     // TODO: development assertion or warning if not already normalized
-    let index = fullName.indexOf(':');
-    let type = fullName.substring(0, index)
-    let name = fullName.substring(index + 1, fullName.length);
-    let moduleName = this.namespace.modulePrefix + '/';
+    let prefix, type, name, moduleName;
+
+    const fullNameParts = fullName.split('@');
+
+    // HTMLBars uses helper:@content-helper which collides
+    // with ember-cli namespace detection.
+    // This will be removed in a future release of HTMLBars.
+    if (fullNameParts.length === 2 && fullName !== 'helper:@content-helper') {
```

Comment:
maybe the split can be made more intelligent, but hardcoding ^ seems problematic.

I believe ^ relies on the fact that `:@` is not a valid namespace, so our split likely needs to be more aware of this as well.